src/orca/services/nextflowtower/client.py

```diff
@@ -131,56 +125,205 @@ def get(self, path: str, **kwargs) -> dict[str, Any]:
+                objects.append(org)
+        return objects
+
+    def list_user_workspaces(self) -> list[models.Workspace]:
         """List the workspaces that are available to the current user.
 
         Returns:
             List of user workspaces.
         """
         user = self.get_user_info()
-        orgs_and_workspaces = self.get_user_workspaces_and_orgs(user["id"])
-
-        workspaces = list()
-        for workspace in orgs_and_workspaces:
-            # Response includes organizations, which don't have workspace IDs
-            if workspace["workspaceId"] is None:
-                continue
-            workspaces.append(workspace)
-        return workspaces
+        items = self.list_user_workspaces_and_orgs(user.id)
+        return [item for item in items if isinstance(item, models.Workspace)]
+
+    def generate_params(
+        self,
+        workspace_id: Optional[int],
+        **kwargs,
+    ) -> dict[str, Any]:
+        """Generate URL query parameters.
+
+        Args:
+            workspace_id: Tower workspace ID.
+            **kwargs: Additional query parameters that are included
+                if they are not set to None.
+
+        Returns:
+            URL query parameters based on input.
+        """
+        params = {}
+        if workspace_id:
+            params["workspaceId"] = int(workspace_id)
+        for name, value in kwargs.items():
+            if value is not None:
+                params[name] = value
+        return params
+
+    def get_compute_env(
+        self,
+        compute_env_id: str,
+        workspace_id: Optional[int] = None,
+    ) -> models.ComputeEnv:
+        """Retrieve information about a given compute environment.
+
+        Args:
+            compute_env_id: Compute environment ID.
+            workspace_id: Tower workspace ID.
+
+        Returns:
+            Compute environment instance.
+        """
+        path = f"/compute-envs/{compute_env_id}"
+        params = self.generate_params(workspace_id, attributes="labels")
+        response = self.get(path, params=params)
+        unwrapped = self.unwrap(response, "computeEnv")
+        return models.ComputeEnv.from_response(unwrapped)
+
+    def list_compute_envs(
+        self,
+        workspace_id: Optional[int] = None,
+        status: Optional[str] = None,
+    ) -> list[models.ComputeEnvSummary]:
+        """List all compute environments.
+
+        Args:
+            workspace_id: Tower workspace ID. Defaults to None.
+            status: Compute environment status to filter on.
+                Defaults to None.
+
+        Returns:
+            List of compute environments.
+        """
+        path = "/compute-envs"
+        params = self.generate_params(workspace_id, status=status)
+        response = self.get(path, params=params)
+        items = self.unwrap(response, "computeEnvs")
+        return [models.ComputeEnvSummary.from_response(item) for item in items]
+
+    def create_label(
```

Comment:
I've been able to create labels with my personal user, which has the "launcher" role. This would be the minimum level of access needed by Orca if you want to launch workflows. If you try to launch a workflow with a read-only token, you should run into an unauthorized exception.